backend/src/main/java/com/htecgroup/skynest/service/impl/FolderServiceImpl.java

```diff
@@ -136,6 +138,33 @@ public List<FolderResponse> getAllFoldersWithParent(UUID parentFolderId) {
     return asFolderResponseList(allFolders);
   }
 
+  @Override
+  public void moveFolderToRoot(UUID folderId) {
+    FolderEntity folderEntity =
+        folderRepository.findById(folderId).orElseThrow(FolderNotFoundException::new);
+    if (folderEntity.getParentFolder() == null) {
+      throw new FolderAlreadyInsideBucketException();
+    }
+    folderEntity.setParentFolder(null);
+    folderRepository.save(folderEntity);
+    actionService.recordAction(Collections.singleton(folderEntity), ActionType.MOVE);
+  }
+
+  @Override
+  public void moveFolderToFolder(UUID folderId, UUID destinationFolderId) {
+    FolderEntity folderEntity =
+        folderRepository.findById(folderId).orElseThrow(FolderNotFoundException::new);
+    FolderEntity parentFolderEntity =
+        folderRepository.findById(destinationFolderId).orElseThrow(FolderNotFoundException::new);
+    if (folderEntity.getParentFolder() != null
+        && folderEntity.getParentFolder().getId() == parentFolderEntity.getId()) {
+      throw new FolderAlreadyInsideFolderException();
+    }
+    folderEntity.setParentFolder(parentFolderEntity);
```

Comment:
what if the destination folder is a child folder?
I don't think it makes sense to move a folder into a child folder
or rather, the destination must not be inside of the folder being moved

Reply:
Illusion

Reply:
@DarkoVeljanoskiHtec something like this - for the destination folder, iterate over all parents until parent is not null, and check whether any parent is the folder you're moving?

Reply:
like in [this PR](https://github.com/JDP2022SkyNest/SkyNest/pull/207)